scanpipe/pipes/d2d.py

```diff
@@ -387,3 +421,192 @@ def purldb_match(project, extensions, logger=None):
+    resource_count = from_resources.count()
+
+    if logger:
+        logger(
+            f"Mapping {resource_count:,d} from/ resources using javascript match "
+            f"against to/ codebase"
+        )
+
+    resource_iterator = from_resources.iterator(chunk_size=2000)
+    last_percent = 0
+    start_time = timer()
+    for resource_index, from_resource in enumerate(resource_iterator):
+        last_percent = pipes.log_progress(
+            logger,
+            resource_index,
+            resource_count,
+            last_percent,
+            increment_percent=10,
+            start_time=start_time,
+        )
+        _resource_javascript_match(from_resource, to_resources)
+
+
+def _resource_javascript_match(from_resource, to_resources):
+    path = Path(from_resource.path.lstrip("/"))
+    if path.name.startswith("."):
+        return
+
+    basename, from_extension = _get_basename_and_extension(path.name)
+    path_parts = (path.parent / basename).parts
+    path_parts_len = len(path_parts)
+
+    for index in range(1, path_parts_len - 1):
+        current_parts = path_parts[index:]
+        current_path = "/".join(current_parts)
+
+        any_match = False
+
+        for candidate in PROSPECTIVE_JAVASCRIPT_MATCH.get(from_extension, []):
+            match_type = "path"
+
+            query = Q(
+                *[
+                    Q(path__endswith=f"{current_path}{extension}")
+                    for extension in candidate["to_related"]
+                ],
+                _connector=Q.OR,
+            )
+
+            matches = to_resources.filter(query)
+
+            if not matches:
+                continue
+
+            if len(matches) > len(current_parts) + len(candidate["to_related"]):
+                continue
+
+            if _is_minified_and_map_compiled_from_source(
+                matches,
+                from_resource,
+                minified_extension=candidate["to_minified_ext"],
+            ):
+                match_type = "js_compiled"
+
+            for match in matches:
+                any_match = True
+
+                relation = CodebaseRelation.objects.filter(
+                    from_resource=from_resource,
+                    to_resource=match,
+                    map_type=match_type,
+                )
+                if not relation.exists():
+                    pipes.make_relation(
+                        from_resource=from_resource,
+                        to_resource=match,
+                        map_type=match_type,
+                        extra_data={
+                            "path_score": f"{len(current_parts)}/{path_parts_len-1}",
+                        },
+                    )
+
+        if any_match:
+            break
+
+
+def _is_minified_and_map_compiled_from_source(
+    to_resources, from_source, minified_extension
+):
+    """Check if a minified file and its map were compiled from a source file."""
+    if not minified_extension:
```

Comment:
What is this `minified_extension`?

Reply:
File extension of minified files like .js, .css